filecoin-proofs/src/api/internal.rs

```diff
@@ -416,7 +402,13 @@ fn verify_post_fixed_sectors_count(
     let commitments = fixed
         .comm_rs
         .iter()
-        .map(|comm_r| PedersenDomain(bytes_into_fr::<Bls12>(comm_r).unwrap().into_repr()))
+        .map(|comm_r| {
+            PedersenDomain(
+                bytes_into_fr::<Bls12>(comm_r)
+                    .expects("could not could not map comm_r to Fr")
```

Comment:
@laser This is not good to panic here, as this means an invalid `comm_r` triggers a panic. Instead this should simply fail the verification.

Reply:
> This is not good to panic here

@porcuquine [does not agree with you](https://filecoinproject.slack.com/archives/CEGB67XJ8/p1552509403191100?thread_ts=1552509239.190000):

> This is a 100% should-panic.

Reply:
@porcuquine we should talk^^

Reply:
Nah, this can/should return an error. When this came up before I spoke too soon and made a wrong assumption about where this was coming from. My mistake — sorry for the noise.

Reply:
Without the context, I mistakenly assumed this was caused by a bug in the challenge-seed truncating code. A`BadFrBytes` error there would have signified a programmer error (mine), and the right answer would have been to correct the code.

Reply:
I do think this should use the error-reporting mechanism to inform the API caller that malformed input was received — since that is almost certainly the result of a bug somewhere else. If we can't return an error and can only fail verification, obviously we need to fail rather than report successful verification on bad input.